### PR TITLE
[OCI TF] administrator users are associated with api key

### LINKED ISSUE #1976: Add OCI TF check - OCI tenancy administrator users are associated with API keys
Equivalent RQL query

config from cloud.resource where cloud.type = 'oci' AND api.name = 'oci-iam-group' as X; config from cloud.resource where api.name = 'oci-iam-user' as Y; filter '($.X.name equals Administrators) and ($.X.groupMembers[*].userId contains $.Y.id) and ($.Y.apiKeys[*] size greater than 0)';show Y;

### PR DESCRIPTION
**Pull request by Eliran-Turgeman**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

resolves #1976 

Added an OCI terraform check that will enforce that administrator users will be associated with an API key.

* I do need an ID to assign to this check

### CODE REVIEW

File: checkov/terraform/checks/graph_checks/oci/AdministratorUserAssociatedWithAPIKey.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+metadata:
+  id: "PLACEHOLDER"
```
Comment (nimrodkor) [inline checkov/terraform/checks/graph_checks/oci/AdministratorUserAssociatedWithAPIKey.yaml:2]: @Eliran-Turgeman doesn't sound like the correct ID to me 😆

**Review by nimrodkor — COMMENTED** [review]
Very nice @Eliran-Turgeman !

File: checkov/terraform/checks/graph_checks/oci/AdministratorUserAssociatedWithAPIKey.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          - oci_identity_group
+        operator: not_equals
+        value: "Administrators"
+    - and:
+      - resource_types:
+          - oci_identity_user_group_membership
+        connected_resource_types:
+          - oci_identity_user
+        operator: not_exists
+        cond_type: connection
```
Comment (gruebel) [inline checkov/terraform/checks/graph_checks/oci/AdministratorUserAssociatedWithAPIKey.yaml:59]:
```suggestion
    - resource_types:
        - oci_identity_user_group_membership
      connected_resource_types:
        - oci_identity_user
      operator: not_exists
      cond_type: connection
```
the `and` is not needed here 🙂

Comment (gruebel) [reply checkov/terraform/checks/graph_checks/oci/AdministratorUserAssociatedWithAPIKey.yaml:59]: you could also merge the other two `and` blocks to one and then do some differentiation in a nested `or` block, but I think your approach is easier to understand 👍

**Review by gruebel — COMMENTED** [review]
nice work 🚀

File: checkov/terraform/checks/graph_checks/oci/AdministratorUserAssociatedWithAPIKey.yaml (hunk 1 above)
Comment (Eliran-Turgeman) [reply checkov/terraform/checks/graph_checks/oci/AdministratorUserAssociatedWithAPIKey.yaml]: Changed to CKV2_OCI_1

File: checkov/terraform/checks/graph_checks/oci/AdministratorUserAssociatedWithAPIKey.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
+metadata:
+  id: "PLACEHOLDER"
+  name: "Ensure administrator users are associated with API keys"
+  category: "IAM"
+scope:
+  provider: 'oci'
```
[the remark this answers is not in the document]
Comment (Eliran-Turgeman) [reply checkov/terraform/checks/graph_checks/oci/AdministratorUserAssociatedWithAPIKey.yaml]: Removed

File: checkov/terraform/checks/graph_checks/oci/AdministratorUserAssociatedWithAPIKey.yaml (hunk 2 above)
Comment (Eliran-Turgeman) [reply checkov/terraform/checks/graph_checks/oci/AdministratorUserAssociatedWithAPIKey.yaml]: Deleted the unnecessary `and` block

**Comment by Eliran-Turgeman** [comment]
Note that I changed the check logic to fail users that are admins and associated with api keys (after clarifying the requirements with Gilad)

**Review by nimrodkor — APPROVED** [review]
GJ!

**Pull request merged**